This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marco Borrini <borrinimarco74@gmail.com>
Date: Tue, 4 Aug 2026 16:07:00 +0200
Subject: [PATCH 1/1] Fix static build missing dependencies and linker order

KF6NewStuffCore links privately to the internal convenience library
knscore_jobs_static. In a static build, this requires knscore_jobs_static
to be exported alongside KF6NewStuffCore in its EXPORT set so that CMake
can resolve the target during linking of consumers.

Also, knewstuff_qml_STATIC explicitly links Qt6::Qml but uses classes
from Qt6Quick. Because it didn't explicitly link Qt6::Quick, the static
linker command line evaluated Qt6::Qml *before* Qt6::Quick, leading to
undefined references inside Qt6Quick for Qml symbols. Explicitly adding
Qt6::Quick to its dependencies fixes the linker ordering in static builds.

Additionally, KF6NewStuffWidgets links to Qt6QuickWidgets, so this
must be advertised in KF6NewStuffConfig.cmake.in for static builds.

diff -ur knewstuff-6.28.0/KF6NewStuffConfig.cmake.in knewstuff-6.28.0-new/KF6NewStuffConfig.cmake.in
--- knewstuff-6.28.0/KF6NewStuffConfig.cmake.in	2026-07-03 12:57:46.000000000 +0200
+++ knewstuff-6.28.0-new/KF6NewStuffConfig.cmake.in	2026-08-04 16:48:01.143794463 +0200
@@ -11,6 +11,7 @@
 if (NOT @BUILD_SHARED_LIBS@)
     find_dependency(Qt6Qml @REQUIRED_QT_VERSION@)
     find_dependency(Qt6Quick @REQUIRED_QT_VERSION@)
+    find_dependency(Qt6QuickWidgets @REQUIRED_QT_VERSION@)
 
     find_dependency(KF6I18n "@KF_DEP_VERSION@")
 endif()
diff -ur knewstuff-6.28.0/src/core/CMakeLists.txt knewstuff-6.28.0-new/src/core/CMakeLists.txt
--- knewstuff-6.28.0/src/core/CMakeLists.txt	2026-07-03 12:57:46.000000000 +0200
+++ knewstuff-6.28.0-new/src/core/CMakeLists.txt	2026-08-04 16:07:53.041334111 +0200
@@ -149,7 +149,7 @@
   REQUIRED_HEADERS KNewStuffCore_HEADERS
 )
 
-install(TARGETS KF6NewStuffCore EXPORT KF6NewStuffCoreTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS KF6NewStuffCore knscore_jobs_static EXPORT KF6NewStuffCoreTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(
     FILES
diff -ur knewstuff-6.28.0/src/qtquick/CMakeLists.txt knewstuff-6.28.0-new/src/qtquick/CMakeLists.txt
--- knewstuff-6.28.0/src/qtquick/CMakeLists.txt	2026-07-03 12:57:46.000000000 +0200
+++ knewstuff-6.28.0-new/src/qtquick/CMakeLists.txt	2026-08-04 16:18:43.636126053 +0200
@@ -26,6 +26,7 @@
 target_link_libraries(knewstuff_qml_STATIC PUBLIC
     Qt6::Core
     Qt6::Gui # QImage
+    Qt6::Quick
     Qt6::Qml
     KF6::ConfigCore
     KF6::I18n
@@ -76,7 +77,7 @@
     qml/private/entrygriddelegates/TileDelegate.qml
 )
 
-target_link_libraries (newstuffqmlplugin PRIVATE knewstuff_qml_STATIC)
+target_link_libraries (newstuffqmlplugin PRIVATE knewstuff_qml_STATIC Qt6::Qml Qt6::QmlModels)
 
 ecm_finalize_qml_module(newstuffqmlplugin DESTINATION ${KDE_INSTALL_QMLDIR})
 
diff -ur knewstuff-6.28.0/src/tools/knewstuff-dialog/CMakeLists.txt knewstuff-6.28.0-new/src/tools/knewstuff-dialog/CMakeLists.txt
--- knewstuff-6.28.0/src/tools/knewstuff-dialog/CMakeLists.txt	2026-07-03 12:57:46.000000000 +0200
+++ knewstuff-6.28.0-new/src/tools/knewstuff-dialog/CMakeLists.txt	2026-08-04 16:21:07.556710776 +0200
@@ -14,8 +14,8 @@
     Qt6::Core
     Qt6::Gui
     Qt6::Widgets
-    Qt6::Qml
     Qt6::Quick
+    Qt6::Qml
     KF6::ConfigCore
     KF6::I18n
     KF6::NewStuffCore
